File: clojure/armstrong-numbers/src/armstrong_numbers.clj
(ns armstrong-numbers)

(defn explode [num]
  (map #(- (int %) (int \0)) (str num)))

(defn exp [base power]
  (loop [total 1 p power]
   (if (zero? p) total
    (recur (*' total base) (dec p)))))

(defn armstrong? [num]
  (let [digit (explode num) pow (count digit)]
    (= num (reduce + (map #(exp % pow) digit)))))

File: go/armstrong-numbers/armstrong_numbers.go
package armstrong

import (
	"math"
	"strconv"
)

func IsNumber(n int) bool {
	number := strconv.Itoa(n)
	power := len(number)

	var sum int
	for _, num := range number {
		digit := num - '0'
		sum += int(math.Pow(float64(digit), float64(power)))
	}

	return sum == n
}
